subsys/bluetooth/controller/ll_sw/lll.h

```diff
@@ -155,94 +155,51 @@ enum node_rx_type {
-#if defined(CONFIG_BT_CTLR_ADV_EXT)
-	NODE_RX_TYPE_EXT_ADV_TERMINATE = 0x0b,
-#endif /* CONFIG_BT_CTLR_ADV_EXT */
-#endif /* CONFIG_BT_BROADCASTER */
-
-#if defined(CONFIG_BT_CTLR_SCAN_REQ_NOTIFY)
-	NODE_RX_TYPE_SCAN_REQ = 0x0c,
-#endif /* CONFIG_BT_CTLR_SCAN_REQ_NOTIFY */
-
-#if defined(CONFIG_BT_CONN)
-	NODE_RX_TYPE_CONNECTION = 0x0d,
-	NODE_RX_TYPE_TERMINATE = 0x0e,
-	NODE_RX_TYPE_CONN_UPDATE = 0x0f,
-	NODE_RX_TYPE_ENC_REFRESH = 0x10,
-
-#if defined(CONFIG_BT_CTLR_LE_PING)
-	NODE_RX_TYPE_APTO = 0x11,
-#endif /* CONFIG_BT_CTLR_LE_PING */
-
-	NODE_RX_TYPE_CHAN_SEL_ALGO = 0x12,
-
-#if defined(CONFIG_BT_CTLR_PHY)
-	NODE_RX_TYPE_PHY_UPDATE = 0x13,
-#endif /* CONFIG_BT_CTLR_PHY */
-
-#if defined(CONFIG_BT_CTLR_CONN_RSSI)
-	NODE_RX_TYPE_RSSI = 0x14,
-#endif /* CONFIG_BT_CTLR_CONN_RSSI */
-#endif /* CONFIG_BT_CONN */
-
-#if defined(CONFIG_BT_CTLR_PROFILE_ISR)
-	NODE_RX_TYPE_PROFILE = 0x15,
-#endif /* CONFIG_BT_CTLR_PROFILE_ISR */
-
-#if defined(CONFIG_BT_CTLR_ADV_INDICATION)
-	NODE_RX_TYPE_ADV_INDICATION = 0x16,
-#endif /* CONFIG_BT_CTLR_ADV_INDICATION */
-
-#if defined(CONFIG_BT_CTLR_SCAN_INDICATION)
-	NODE_RX_TYPE_SCAN_INDICATION = 0x17,
-#endif /* CONFIG_BT_CTLR_SCAN_INDICATION */
-
-#if defined(CONFIG_BT_CTLR_PERIPHERAL_ISO)
-	NODE_RX_TYPE_CIS_REQUEST = 0x18,
-#endif /* CONFIG_BT_CTLR_PERIPHERAL_ISO */
-
-#if defined(CONFIG_BT_CTLR_CENTRAL_ISO) || \
-	defined(CONFIG_BT_CTLR_PERIPHERAL_ISO)
-	NODE_RX_TYPE_CIS_ESTABLISHED = 0x19,
-#endif /* CONFIG_BT_CTLR_CENTRAL_ISO || CONFIG_BT_CTLR_PERIPHERAL_ISO */
-
-#if defined(CONFIG_BT_HCI_MESH_EXT)
-	NODE_RX_TYPE_MESH_ADV_CPLT = 0x1a,
-	NODE_RX_TYPE_MESH_REPORT = 0x1b,
-#endif /* CONFIG_BT_HCI_MESH_EXT */
+	NODE_RX_TYPE_REPORT,
+	NODE_RX_TYPE_EXT_1M_REPORT,
+	NODE_RX_TYPE_EXT_2M_REPORT,
+	NODE_RX_TYPE_EXT_CODED_REPORT,
+	NODE_RX_TYPE_EXT_SCAN_TERMINATE,
+	NODE_RX_TYPE_SYNC,
+	NODE_RX_TYPE_SYNC_LOST,
+	NODE_RX_TYPE_EXT_ADV_TERMINATE,
+	NODE_RX_TYPE_SCAN_REQ,
+	NODE_RX_TYPE_CONNECTION,
+	NODE_RX_TYPE_TERMINATE,
+	NODE_RX_TYPE_CONN_UPDATE,
+	NODE_RX_TYPE_ENC_REFRESH,
+	NODE_RX_TYPE_APTO,
+	NODE_RX_TYPE_CHAN_SEL_ALGO,
+	NODE_RX_TYPE_PHY_UPDATE,
+	NODE_RX_TYPE_RSSI,
+	NODE_RX_TYPE_PROFILE,
+	NODE_RX_TYPE_ADV_INDICATION,
+	NODE_RX_TYPE_SCAN_INDICATION,
+	NODE_RX_TYPE_CIS_REQUEST,
+	NODE_RX_TYPE_CIS_ESTABLISHED,
+	NODE_RX_TYPE_MESH_ADV_CPLT,
+	NODE_RX_TYPE_MESH_REPORT,
+
+	/* No entries shall be added after NODE_RX_TYPE_USER_START */
+	NODE_RX_TYPE_USER_START,
+};
 
-/* Following proprietary defines must be at end of enum range */
 #if defined(CONFIG_BT_CTLR_USER_EXT)
-	NODE_RX_TYPE_USER_START = 0x1c,
-	NODE_RX_TYPE_USER_END = NODE_RX_TYPE_USER_START +
+/* The NODE_RX_TYPE_USER_END is defined outside the enum to comply with MISRA:
+ * MISRA C:2004, 9.3 - In an enumerator list, the "=" construct shall not be
+ * used to explicitly initialize members other than the first, unless all items
+ * are explicitly initialized.
+ */
+#define NODE_RX_TYPE_USER_END = NODE_RX_TYPE_USER_START +
```

Comment:
The '=' should be removed for this to even compile.

I understand what you are trying to do, but it gives warnings when compiling with CONFIG_BT_CTLR_USER_EXT=y as NODE_RX_TYPE_USER_END is now no longer a value in the enumerated type.

warning: case value ‘X’ not in enumerated type ‘enum node_rx_type’ [-Wswitch]                                         
case NODE_RX_TYPE_USER_START ... NODE_RX_TYPE_USER_END:

Not sure how i like those warnings, could be a problem if you were dedicated to the -Werror style of life...

Reply:
I have an idea on how to solves this: we need a pre-processor enum value generator thingy...

As `CONFIG_BT_CTLR_USER_EVT_RANGE` has the range 0 to 10 the following construction can genrate the padding need between `NODE_RX_TYPE_USER_START` and `NODE_RX_TYPE_USER_END`

It's not entirely the same as `NODE_RX_TYPE_USER_START` != `NODE_RX_TYPE_USER_DUMMY_1` and `NODE_RX_TYPE_USER_END` != `NODE_RX_TYPE_USER_DUMMY_5` if `CONFIG_BT_CTLR_USER_EVT_RANGE=5`, but that should be ok.

```
#define NODE_RX_TYPE_USER_GENERATOR_0(X)
#define NODE_RX_TYPE_USER_GENERATOR_1(X)	 X ## 1,
#define NODE_RX_TYPE_USER_GENERATOR_2(X)	 NODE_RX_TYPE_USER_GENERATOR_1(X) X ## 2,
#define NODE_RX_TYPE_USER_GENERATOR_3(X)	 NODE_RX_TYPE_USER_GENERATOR_2(X) X ## 3,
#define NODE_RX_TYPE_USER_GENERATOR_4(X)	 NODE_RX_TYPE_USER_GENERATOR_3(X) X ## 4,
#define NODE_RX_TYPE_USER_GENERATOR_5(X)	 NODE_RX_TYPE_USER_GENERATOR_4(X) X ## 5,
#define NODE_RX_TYPE_USER_GENERATOR_6(X)	 NODE_RX_TYPE_USER_GENERATOR_5(X) X ## 6,
#define NODE_RX_TYPE_USER_GENERATOR_7(X)	 NODE_RX_TYPE_USER_GENERATOR_6(X) X ## 7,
#define NODE_RX_TYPE_USER_GENERATOR_8(X)	 NODE_RX_TYPE_USER_GENERATOR_7(X) X ## 8,
#define NODE_RX_TYPE_USER_GENERATOR_9(X)	 NODE_RX_TYPE_USER_GENERATOR_8(X) X ## 9,
#define NODE_RX_TYPE_USER_GENERATOR_10(X)	 NODE_RX_TYPE_USER_GENERATOR_9(X) X ## 10,
#define NODE_RX_TYPE_USER_GENERATOR_EXPAND(N, X) NODE_RX_TYPE_USER_GENERATOR_##N(X)
#define NODE_RX_TYPE_USER_GENERATOR(N, X)	 NODE_RX_TYPE_USER_GENERATOR_EXPAND(N, X)
```

```
enum node_rx_type {
...
	NODE_RX_TYPE_USER_START,
	NODE_RX_TYPE_USER_GENERATOR(CONFIG_BT_CTLR_USER_EVT_RANGE, NODE_RX_TYPE_USER_DUMMY_)
	NODE_RX_TYPE_USER_END,
};
```

Reply:
Good catch! I guess CI doesn't have a build config with `CONFIG_BT_CTLR_USER_EXT=y` since it didn't catch this... 

I understand your worry, and I would really like to fix it. However for MISRA we cannot have it at the end of the enumeration either. Not sure how big a deal conforming to MISRA is. 

When I build with `west`, I actually get an error rather than a warning.

Reply:
There actually is a "compile test" in tests/bluetooth/ctrl_user_ext that uses `CONFIG_BT_CTLR_USER_EXT=y`, but i guess CI doesn't compile that one.

Reply:
There is `UTIL_LISTIFY()` macro that does something like your generator. Can't you just do it like this (not tested)

```
#define DEFINE_NODE_RX_USER_TYPE(i, _) NODE_RX_TYPE##i,
enum node_rx_type {
   NODE_RX_TYPE_USER_START,
   UTIL_LISTIFY(CONFIG_BT_CTLR_USER_EVT_RANGE, DEFINE_NODE_RX_USER_TYPE, _)
   NODE_RX_TYPE_USER_END,
};
```

Reply:
Was about to say the same as @jukkar. I'll try to see if I can cook up a version using that, as that would be preferable in any way I suppose :)